src/org/exist/storage/NativeBroker.java

```diff
@@ -2021,6 +2021,9 @@ public DocumentImpl getResourceById(final int collectionId, final byte resourceT
             //get the resource uri
             final Value key = new CollectionStore.DocumentKey(collectionId, resourceType, documentId);
             final VariableByteInput vbi = collectionsDb.getAsStream(key);
+            if (vbi == null) {
+                return null;
+            }
             vbi.readInt(); //skip doc id
```

Comment:
This is the actual code that throws. So we must return null beforehand.

Reply:
Hmm... doesn't a null here imply that the collections.dbx has become corrrupted?